pkgs/development/libraries/libxkbcommon/default.nix

```diff
@@ -21,17 +20,15 @@ stdenv.mkDerivation rec {
 
   outputs = [ "out" "dev" "doc" ];
 
-  nativeBuildInputs = [ meson ninja pkg-config yacc doxygen ]
-    ++ lib.optional withWaylandSupport wayland;
-  buildInputs = [ xkeyboard_config libxcb libxml2 ]
-    ++ lib.optionals withWaylandSupport [ wayland wayland-protocols ];
+  nativeBuildInputs = [ meson ninja pkg-config yacc doxygen ];
+  buildInputs = [ xkeyboard_config libxcb libxml2 wayland wayland-protocols ];
```

Comment:
Why did you change that part? This wasn't an accident, `wayland` must be in both `nativeBuildInputs` (wayland-scanner) and `buildInputs` (libs).

Reply:
Everything builds fine here without `nativeBuildInputs` - (with and without the `withWaylandSupport` flag). How would you reproduce the problem?

Reply:
Reason for asking - I"ll add in a comment here to explain the situation and avoid the next person removing it if it is truly needed.

Reply:
> Everything builds fine here without nativeBuildInputs

That's an unfortunate issue for backward compatibility / legacy reasons (`buildInputs` will be added to `PATH` but that shouldn't be the case). See also https://nixos.org/manual/nixpkgs/unstable/#variables-specifying-dependencies, etc.

> How would you reproduce the problem?

E.g. by building `pkgsCross.aarch64-multiplatform.libxkbcommon`.